### ISSUE TITLE
fix false negatives in explicit_outlives_requirements 

### ISSUE THREAD

**Issue by zackmdavis** [rust-lang/rust#54630]
Niko Matsakis [points out that](https://github.com/rust-lang/rust/pull/53013#discussion_r221053173) the explicit_outlives_requirements lint (from #53013, expected to land soon) should also fire on lifetime-outlives bounds and associated-type-outlives bounds, as illustrated by the following two examples:

```rust
struct Foo<'a, 'b: 'a> {
    x: &'a &'b u32
}

struct Bar<'a, T: Iterator> where T::Item: 'a {
    item: &'a T::Item,
}
```

**Comment by fmease** [comment]
We do emit `explicit_outlives_requirements` for the first case (obviously only if the lint level is set to at least warn) but still not for the second.
